lib/pengine/unpack.c

```diff
@@ -3532,142 +3527,150 @@ remap_because(int *i, const char **why, int value, const char *reason)
  * \note This may update the resource's current and next role.
  */
 static void
-remap_operation(xmlNode *xml_op, pe_resource_t *rsc, const pe_node_t *node,
-                pe_working_set_t *data_set, enum action_fail_response *on_fail,
-                int target_rc, int *rc, int *status)
+remap_operation(struct action_history *history,
+                enum action_fail_response *on_fail)
 {
     bool is_probe = false;
-    int orig_exit_status = *rc;
-    int orig_exec_status = *status;
+    int orig_exit_status = history->exit_status;
+    int orig_exec_status = history->execution_status;
     const char *why = NULL;
-    const char *task = crm_element_value(xml_op, XML_LRM_ATTR_TASK);
-    const char *key = get_op_key(xml_op);
-    const char *exit_reason = crm_element_value(xml_op,
-                                                XML_LRM_ATTR_EXIT_REASON);
+    const char *task = history->task;
     char *last_change_s = NULL;
 
     if (pcmk__str_eq(task, CRMD_ACTION_STATUS, pcmk__str_none)) {
         // Remap degraded results to their usual counterparts
-        *rc = pcmk__effective_rc(*rc);
-        if (*rc != orig_exit_status) {
+        history->exit_status = pcmk__effective_rc(history->exit_status);
+        if (history->exit_status != orig_exit_status) {
             why = "degraded monitor result";
-            if (!node->details->shutdown || node->details->online) {
-                record_failed_op(xml_op, node, rsc, data_set);
+            if (!history->node->details->shutdown
+                || history->node->details->online) {
+                record_failed_op(history->xml, history->node, history->rsc,
+                                 history->rsc->cluster);
             }
         }
     }
 
-    if (!pe_rsc_is_bundled(rsc) && pcmk_xe_mask_probe_failure(xml_op)) {
-        if ((*status != PCMK_EXEC_DONE) || (*rc != PCMK_OCF_NOT_RUNNING)) {
-            *status = PCMK_EXEC_DONE;
-            *rc = PCMK_OCF_NOT_RUNNING;
-            why = "irrelevant probe result";
-        }
+    if (!pe_rsc_is_bundled(history->rsc)
```

Comment:
I think it's useful when reading it to know that's happening, and they're in different functions so it would be awkward to add another argument for a trivial gain in efficiency

Reply:
It wouldn't require an extra argument. It would just be something like
```
static inline bool
maskable_probe_failure(struct action_history *history)
{
    return !pe_rsc_is_bundled(history->rsc)
           && pcmk_xe_mask_probe_failure(history->xml);
}
```

and then calling `maskable_probe_failure(history)` in the four places where we currently do both checks together in `unpack.c`.

Reply:
again doesn't matter to me though